### PR TITLE
feat(ads): expand suppression options

### PR DESCRIPTION
**Pull request by miguelpeixe**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Removes suppression config backend from this plugin and adjust wizard options for https://github.com/Automattic/newspack-ads/pull/482

### How to test the changes in this Pull Request:

Visit https://github.com/Automattic/newspack-ads/pull/482 for further instructions.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: assets/wizards/advertising/views/suppression/index.js [hunk 1]
```diff
@@ -1,80 +1,132 @@
...
+		} ).then( result => {
+			setPostTypes(
+				Object.values( result )
+					.filter( postType => postType.viewable === true && postType.visibility?.show_ui === true )
+					.map( postType => ( {
+						value: postType.slug,
+						label: postType.name,
+					} ) )
+			);
+		} );
```
Comment (dkoo) [inline assets/wizards/advertising/views/suppression/index.js:41]: No error handling?

File: assets/wizards/advertising/views/suppression/index.js [hunk 2]
```diff
@@ -1,80 +1,132 @@
...
+						} else {
+							newPostTypes = newPostTypes.filter( type => type !== postType.value );
+						}
+						setConfig( { ...config, post_types: newPostTypes } );
+					} }
+				/>
+			) ) }
+			<SectionHeader
+				title={ __( 'Tags', 'newspack' ) }
+				description={ __( 'Suppress ads on specific tags and its archive pages.', 'newspack' ) }
```
Comment (dkoo) [inline assets/wizards/advertising/views/suppression/index.js:84]:
```suggestion
				description={ __( 'Suppress ads on specific tags and their archive pages.', 'newspack' ) }
```

File: assets/wizards/advertising/views/suppression/index.js [hunk 3]
```diff
@@ -1,80 +1,132 @@
...
 				onChange={ selected => {
-					onChange( {
+					setConfig( {
 						...config,
-						specific_tag_archive_pages: selected.map( item => item.id ),
+						tags: selected.map( item => item.id ),
 					} );
 				} }
-				label={ __( 'Specific tags archive pages', 'newspack ' ) }
+				label={ __( 'Tags to suppress ads on its archive page and posts', 'newspack ' ) }
```
Comment (dkoo) [inline assets/wizards/advertising/views/suppression/index.js:94]:
```suggestion
				label={ __( 'Tags to suppress ads on (archives and posts)', 'newspack ' ) }
```

File: assets/wizards/advertising/views/suppression/index.js [hunk 4]
```diff
@@ -1,80 +1,132 @@
...
 				} }
 				label={ __( 'All tag archive pages', 'newspack' ) }
 			/>
-
 			<SectionHeader
-				title={ __( 'Category Archive Pages', 'newspack' ) }
+				title={ __( 'Categories', 'newspack' ) }
 				description={ __(
-					'Suppress ads on automatically generated pages displaying a list of posts of a category.',
+					'Suppress ads on specific categories and its archive pages.',
```
Comment (dkoo) [inline assets/wizards/advertising/views/suppression/index.js:108]:
```suggestion
					'Suppress ads on specific categories and their archive pages.',
```

File: assets/wizards/advertising/views/suppression/index.js [hunk 5]
```diff
@@ -1,80 +1,132 @@
...
 				onChange={ selected => {
-					onChange( {
+					setConfig( {
 						...config,
-						specific_category_archive_pages: selected.map( item => item.id ),
+						categories: selected.map( item => item.id ),
 					} );
 				} }
-				label={ __( 'Specific category archive pages', 'newspack ' ) }
+				label={ __( 'Categories to suppress ads on its archive page and posts', 'newspack ' ) }
```
Comment (dkoo) [inline assets/wizards/advertising/views/suppression/index.js:120]:
```suggestion
				label={ __( 'Categories to suppress ads on (archives and posts)', 'newspack ' ) }
```

File: assets/wizards/advertising/views/suppression/index.js [hunk 6]
```diff
@@ -1,80 +1,132 @@
...
+	};
+	const updateConfig = () => {
+		setInFlight( true );
+		apiFetch( {
+			path: '/newspack-ads/v1/suppression',
+			method: 'POST',
+			data: { config },
+		} )
+			.then( setConfig )
+			.finally( () => {
```
Comment (dkoo) [inline assets/wizards/advertising/views/suppression/index.js:51]: No error handling?

**Review by dkoo — CHANGES_REQUESTED** [review]
A few nitpicks below!

File: assets/wizards/advertising/views/suppression/index.js [hunk 7]
```diff
@@ -1,80 +1,132 @@
...
-				title={ __( 'Tag Archive Pages', 'newspack' ) }
-				description={ __(
-					'Suppress ads on automatically generated pages displaying a list of posts with a tag.',
-					'newspack'
-				) }
+				title={ __( 'Post Types', 'newspack' ) }
+				description={ __( 'Suppress ads on specific post types.', 'newspack' ) }
+			/>
+			{ postTypes.map( postType => (
+				<ToggleControl
```
[the remark this answers is not in the document]
Comment (thomasguillot) [reply assets/wizards/advertising/views/suppression/index.js:67]: I was going to suggest checkboxes and grid too.

> And/or perhaps a control to select or deselect all?

This is a great idea.

File: assets/wizards/advertising/views/suppression/index.js (hunk 1 above)
Comment (miguelpeixe) [reply assets/wizards/advertising/views/suppression/index.js]: Fixed in 46f4755acee4a213a66580ba287cc1c7bc48a3b5

File: assets/wizards/advertising/views/suppression/index.js (hunk 6 above)
Comment (miguelpeixe) [reply assets/wizards/advertising/views/suppression/index.js:58]: Fixed in 46f4755acee4a213a66580ba287cc1c7bc48a3b5

File: assets/wizards/advertising/views/suppression/index.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (miguelpeixe) [reply assets/wizards/advertising/views/suppression/index.js]: Good call! It does look a lot better:

image

Done in 6b45bb1e5332db100064e465a411a2f058034fe0

[the remark this answers is not in the document]
Comment (dkoo) [reply assets/wizards/advertising/views/suppression/index.js]: This looks better! Unfortunately Gutenberg checkboxes don't seem to support the standard shift-click behavior to select or deselect multiple checkboxes. Also I still think a control to select or deselect all would be good here, but it's non-blocking.

**Review by dkoo — APPROVED** [review]
Looking good with the new checkboxes + columns 👍

**Pull request merged**